### PR TITLE
Health of PDS services in case of failures

### PR DESCRIPTION
**Pull request by Madanagopal19**
PDS System Testcase  - Health of PDS services in case of failures

1. It deploys PDS Data services
2. Deletes the data service pods
3. Validates the health status of the data services pods to be unhealthy
4. Waits for sometime for the data service pods to be healthy

logs - https://gist.github.com/Madanagopal19/296fb3aef1ce1f258584adef4dcafd9f
Branch build test pipeline - https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/PDS%2Fpds-branch-build-test/detail/pds-branch-build-test/87/pipeline

### CODE REVIEW

File: drivers/pds/lib/utils.go [hunk 1]
```diff
@@ -708,6 +718,30 @@ func GetAllVersionsImages(dataServiceID string) (map[string][]string, map[string
...
+	//validate the deployments in pds
+	err = wait.Poll(maxtimeInterval, timeOut, func() (bool, error) {
+		status, res, err := components.DataServiceDeployment.GetDeploymentStatus(deployment.GetId())
+		log.Infof("Health status -  %v", status.GetHealth())
+		if err != nil {
+			log.Errorf("Error occured while getting deployment status %v", err)
+			return false, nil
+		}
+		if res.StatusCode != state.StatusOK {
+			log.Errorf("Error when calling `ApiDeploymentsIdCredentialsGet``: %v\n", err)
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:731]: err will be nil here, yiu ight need to create object and return it

File: drivers/pds/lib/utils.go [hunk 2]
```diff
@@ -708,6 +718,30 @@ func GetAllVersionsImages(dataServiceID string) (map[string][]string, map[string
...
+			log.Errorf("Full HTTP response: %v\n", res)
+			return false, err
+		}
+
+		if !strings.Contains(status.GetHealth(), healthStatus) {
+			return false, nil
+		}
```
Comment (lsrinivas-pure) [inline drivers/pds/lib/utils.go:738]: it would be helpful in logging actual health status as an error

File: drivers/pds/lib/utils.go (hunk 1 above)
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go]: done

**Comment by aravindranath14** [comment]
LGTM

**Review by aravindranath14 — APPROVED** [review]

**Review by dbhatnagar-px — APPROVED** [review]
LGTM - Pls check if the extra wrapper function could be avoided.

File: drivers/pds/lib/utils.go [hunk 3]
```diff
@@ -440,37 +440,47 @@ func GetPods(namespace string) (*corev1.PodList, error) {
...
-		err = k8sOps.ValidatePod(&pod, timeOut, timeInterval)
+		err = k8sCore.ValidatePod(&pod, timeOut, timeInterval)
 		if err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
+// DeleteDeploymentPods deletes the given pods
```
[the remark this answers is not in the document]
Comment (Madanagopal19) [reply drivers/pds/lib/utils.go:475]: I don't want to bring the sched-ops lib in our test cases file, hence wrote that function in our utils.

**Pull request merged**